asttokens/mark_tokens.py

```diff
@@ -88,6 +83,9 @@ def _visit_after_children(self, node, parent_token, token):
     first = token
     last = None
     for child in cast(Callable, self._iter_children)(node):
+      # astroid slices have especially wrong positions, we don't want them to corrupt their parents.
+      if util.is_empty_astroid_slice(child):
```

Comment:
Might it be a better alternative to filter out empty slices in `iter_children_astroid` in `util.py`? (Perhaps that would be wrong, but seems this is done for some nodes already.)

Reply:
Just tried that, it didn't work